packages/web-config-server/src/apiV1/dataBuilders/helpers/groupEvents.js

```diff
@@ -10,29 +10,71 @@ const getOrgUnits = async (models, { parentCode, type, hierarchyId }) => {
   return parentOrgUnit.getDescendantsOfType(hierarchyId, type);
 };
 
+/**
+ * Given an object of eventsByOrgUnitCode, returns the same object with the code changed to name, but
+ * disambiguated, e.g. 'Melbourne', 'Other', 'Other (1)'
+ *
+ * @param allOrgUnits
+ * @param eventsByOrgUnitCode
+ * @returns {{}}
+ * @private
+ */
+const mapOrgUnitCodeToUniqueOrgUnitName = (allOrgUnits, eventsByOrgUnitCode) => {
```

Comment:
There are no cases in production where this is an issue. 

The bug that is being fixed is fixed entirely in the methods `groupByAllOrgUnitNames` and `groupByAllOrgUnitParentNames` by changing to using codes during processing instead of names.

This method simply converts an object keyed by org unit code to an object keyed by org unit name.

The extra work it does to disambiguate non-unique names is done for completeness, as we may run across this problem in the future. And I have added tests to cover this.